### ISSUE TITLE
Colors on map do not match legend

### ISSUE THREAD

**Issue by joellenb** [CEMPD/ctools#190]
The colors on the map should match what the user expects to see based on the ranges in the legend and the values being displayed. Here is an example that I believe shows a clear difference between what the user expects and the colors that are actually shown.
This example is Ships-in-transit (default sources and emissions); Savannah, GA; CO; annual average concentration.

NOTE: Legend has blue between 1E-3 and 1E-2, green between 1E-2 and 1E-1, yellow between 1E-1 and 1E0, and brown greater than 1E0.
Upper-left corner, concentration 0.001276 approx. 1E-3 should be blue; map has green
Top-center, concentration 0.03971 approx. E-2 should be green; map has yellow
Middle, concentration 0.2552 approx. E-1, should be yellow; map has brown
![c-port_sit_colorsnotmatchinglegend](https://cloud.githubusercontent.com/assets/10145737/11043790/72453068-86ec-11e5-8843-09fccf7524d7.PNG)

**Comment by cseppan** [comment]
I suspect this is due to inspect mode displaying the concentration at the nearest receptor, whereas the concentration overlay is based on interpolated values. As a test, try turning on the receptor display, then use inspect mode to click various points near each receptor. I'll also look at switch inspect mode to use averaged values from a set of nearest receptors.

**Comment by joellenb** [comment]
I tried to click on the receptor close to the spot where I had previously clicked (above). Please note the top center value is now 0.04152, approximately E-2, should be green, but is shown as bright yellow.

![c-port_sit_wreceptors_colorsnotmatchinglegend](https://cloud.githubusercontent.com/assets/10145737/11044481/2fc5ab7e-86f0-11e5-8fdf-9ba4f924d0ac.PNG)

**Comment by cseppan** [comment]
What values do you get for the 8 receptors surrounding the 0.04152 value? Just wondering if the receptors closer to the sources (although nothing's very close) are pulling up the average.

**Comment by joellenb** [comment]
Something is odd. The middle value of the 9 is the highest value of the 9 (i.e., is higher than any of the surrounding 8 values).

Here they are:

0.03860                 0.04075                 0.03950
0.03971                 [0.04152]             0.04041
0.03883                 0.04119                 0.04122

When there is a range of values the arithmetic average cannot be at or above the top of the range. The same should hold true for inverse distance weighting (IDW) or a type of kriging. What algorithm is C-PORT using?

Jo Ellen

**Comment by joellenb** [comment]
Here is another example of colors that do not match the legend. I hope that this will help to identify potential problem areas in the code.

I previously suspected that the problems were with values <1 but I didn't have a good example to show that. I think that this one does. This is the annual average concentration for formaldehyde * 10,000,000 emissions for Ships-in-transit only in Savannah, GA. 
NOTE: Legend has purple/blue around 1E-1, green between 1E-1 and 1E0, yellow and brown greater than 1E0.

Middle has concentration >9.8 and matches top color of legend at brown
Upper-left corner, concentration 0.04506 approx. 1E-2 should be clear (below purple); map has green
To the right of upper-left corner, concentration is 0.1369 approx. 1E-1 should be blue; map has yellow

![c-port_sit_form_colorsnotmatchinglegend](https://cloud.githubusercontent.com/assets/10145737/11047576/43d08b32-8701-11e5-84e1-f104374e1632.PNG)

**Comment by cseppan** [comment]
I found some issues with the interpolation code that I've fixed and deployed to treehug.

For testing, I ran the Savannah SIT scenario that Jo Ellen described in the most recent comment. Here's an image showing just the receptor concentrations with no interpolation.

<img width="492" alt="screen shot 2015-11-30 at 3 52 49 pm" src="https://cloud.githubusercontent.com/assets/7329179/11484137/7936fc78-977a-11e5-974e-bb664732ef41.png">

And here's the fixed interpolated image from treehug.

<img width="1082" alt="screen shot 2015-11-30 at 3 52 39 pm" src="https://cloud.githubusercontent.com/assets/7329179/11484138/79370ac4-977a-11e5-8d89-9c0802b189cf.png">
